pkg/controllers/nodetemplate/suite_test.go

```diff
@@ -0,0 +1,365 @@
+/*
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package nodetemplate_test
+
+import (
+	"context"
+	"fmt"
+	"path/filepath"
+	"runtime"
+	"testing"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/aws/awserr"
+	"github.com/aws/aws-sdk-go/service/eventbridge"
+	"github.com/aws/aws-sdk-go/service/sqs"
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+	. "knative.dev/pkg/logging/testing"
+	_ "knative.dev/pkg/system/testing"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	"github.com/aws/karpenter-core/pkg/apis/config/settings"
+	"github.com/aws/karpenter-core/pkg/apis/provisioning/v1alpha5"
+	"github.com/aws/karpenter-core/pkg/operator/injection"
+	"github.com/aws/karpenter-core/pkg/operator/options"
+	. "github.com/aws/karpenter-core/pkg/test/expectations"
+	awssettings "github.com/aws/karpenter/pkg/apis/config/settings"
+	"github.com/aws/karpenter/pkg/apis/v1alpha1"
+	"github.com/aws/karpenter/pkg/controllers/providers"
+	"github.com/aws/karpenter/pkg/errors"
+
+	"github.com/aws/karpenter-core/pkg/test"
+	"github.com/aws/karpenter/pkg/controllers/nodetemplate"
+	awsfake "github.com/aws/karpenter/pkg/fake"
+	awstest "github.com/aws/karpenter/pkg/test"
+)
+
+var ctx context.Context
+var env *test.Environment
+var sqsapi *awsfake.SQSAPI
+var sqsProvider *providers.SQS
+var eventbridgeapi *awsfake.EventBridgeAPI
+var eventBridgeProvider *providers.EventBridge
+var controller *nodetemplate.Controller
+var opts options.Options
+
+var defaultOpts = options.Options{
+	ClusterName:               "test-cluster",
+	ClusterEndpoint:           "https://test-cluster",
+	AWSNodeNameConvention:     string(options.IPName),
+	AWSENILimitedPodDensity:   true,
+	AWSEnablePodENI:           true,
+	AWSDefaultInstanceProfile: "test-instance-profile",
+}
+
+func TestAPIs(t *testing.T) {
+	ctx = TestContextWithLogger(t)
+	RegisterFailHandler(Fail)
+	RunSpecs(t, "AWS Node Template")
```

Comment:
This will be used by FOCUS
```suggestion
	RunSpecs(t, "AWSNodeTemplate")
```